LSPS5/README.md

```diff
@@ -0,0 +1,375 @@
+environment, then when it is suspended, the LSP cannot contact
+the client to perform any operations, such as to forward a
+payment to the client.
+
+Generally, mobile environments will have the mobile OS developers
+run a server or network of servers where an application developer
+can deliver "push notifications".
+When an application developer server contacts the mobile OS
+developer push notification server, the application developer
+server signs and sends a message that is to be displayed to the
+user of the mobile unit, which the mobile OS server will validate
+before causing the message to be displayed to the specified user.
+If the user then taps on this "push notification" on their screen,
+the application is awoken and it can then re-establish its TCP
+connections.
+
+There may be multiple implementations of mobile LSPS-supporting
+clients, each of which run their own application developer server
+and have their own signing keys for these push notifications.
+
+This specification provides a way for a mobile client to
+register some specific webhook by which the LSP can signal a
+push notification to the application developer server, which
+will in turn convert the push notification to one that it itself
+signs and can send to the mobile OS developer server.
+
+### Actors
+
+The 'client' is a specific instance of the client application
+running on some mobile device.
+
+The 'LSP' is a published node that may need to awaken some
+client instance in order to handle some event.
+
+The 'notification delivery service' is the server that a mobile
+client application developer maintains, which is contacted by
+the LSP via an HTTPS `POST` request, in order for the LSP
+to awaken some client instance.
+
+The 'mobile OS server' is the entity that actually allows
+push notifications to be sent to actual mobile devices.
+
+## Protocol
+
+### Webhook Registration API
+
+The client can register a specific webhook for the LSP to
+contact in case the client is asleep, but the LSP needs the
+client to be online and ready to interact.
+
+The client uses the `lsps5.set_webhook` call to specify the
+HTTPS URI that the LSP should `POST` in order to send a
+push notification to the client user.
+It takes the parameters below:
+
+```JSON
+{
+    "webhook": "https://www.example.org/push?l=1234567890abcdefghijklmnopqrstuv&c=best"
+}
+```
+
+`webhook` is a required string, containing the URI of the
+webhook that the LSP can use to push a notification to the
+client.
+
+The client:
+
+* MUST use a HTTPS-protocol URI to the notification delivery
+  service.
+* MUST encode the following data in any part of the URI
+  (hostname, path, or `GET` parameters):
+  * The LSP node ID.
+  * Some authorization to allow the given LSP to wake the
+    client, ideally a proof that the client has a channel
+    or other relationship with the LSP.
+  * Some way to identify the mobile device the client is
+    running on.
```

Comment:
Indeed, the LSP does not impose any format on the URL.  The reason for MUSTiness here is to remind client implementors that they can, and really ought to ("MUST") put the data into the URL instead of requiring additional fields in the LSP->notification server to include that data.

How about "Some way to identify the target device the client is running on"? Desktops and servers are still devices.

Reply:
done, used "target device" more often, with "mobile device" only for specific mobile-device cases.